AccountingService/Controllers/AuthenticationController.cs

```diff
@@ -0,0 +1,123 @@
+    {
+        private SalaryConversionContext salaryConversionContext;
+        public AuthenticationController(SalaryConversionContext context) : base()
+        {
+            salaryConversionContext = context;
+        }
+            [HttpPost("register")]
+        public async Task<ActionResult<SystemUser>> Register(Authentication request)
+        {
+            CreatePasswordHash(request.Password, out byte[] passwordHash, out byte[] passwordSalt);
+            var user = new SystemUser();
+
+            try
+            {
+                const string PasswordRegexPattern = @"^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[^\da-zA-Z]).{8,}$";
+                var useremail= salaryConversionContext.SystemUser.
+                    Where(u => u.Email == request.Email).SingleOrDefault();
+                if (useremail != null)
+                {
+                    return BadRequest("Email already exists.");
+                }
+
+                if (!Regex.IsMatch(request.Password, PasswordRegexPattern))
+                {
+                    return BadRequest("Password must contain at least 8 characters, lowercase and uppercase letters, numbers, and special characters.");
+                }
+                user.Email = request.Email;
+                user.PasswordHash = passwordHash;
+                user.PasswordSalt = passwordSalt;
+                
+                salaryConversionContext.SystemUser.Add(user);
+                await salaryConversionContext.SaveChangesAsync();
+
+                return Ok(user);
+            }
+            catch (Exception )
+            {
+                var error = new Error
+                {
+                    Message= "An error occurred during the registration process",
+                    Code=Code.Connection
+                };
+                return BadRequest(error);
+            }
+        }
+        [HttpPost("login")]
+        public async Task<ActionResult<SystemUser>> Login(Authentication req)
+        {
+            var userlogin = new SystemUser();
+            try
+            {
+                var userdata = await salaryConversionContext.SystemUser.
+                    FirstOrDefaultAsync(u => u.Email == req.Email);
+                if (userdata == null )
+                {
+                    var emailFailed = new Error
+                    {
+                        Message = "Wrong email",
+                        Code = Code.Unknown
+                    };
+                    return BadRequest(emailFailed);
+                }         
+                var verifypassword = VerifyPasswordHash(req.Password, userdata.PasswordHash, userdata.PasswordSalt);
+                if (verifypassword)
+                {
+                    return Ok("You have been successfully logged in");
+                }
+                var error = new Error
+                {
+                    Message = "You have not been successfully logged in",
+                    Code = Code.Connection
```

Comment:
This is not an appropriate code. You should have created a new one for this use case.